pkg/issuer/acme/dns/util/wait.go

```diff
@@ -94,19 +94,22 @@ func checkDNSPropagation(fqdn, value string, nameservers []string) (bool, error)
 
 // checkAuthoritativeNss queries each of the given nameservers for the expected TXT record.
 func checkAuthoritativeNss(fqdn, value string, nameservers []string) (bool, error) {
+	var errs []error
+	var found = false
 	for _, ns := range nameservers {
 		r, err := dnsQuery(fqdn, dns.TypeTXT, []string{net.JoinHostPort(ns, "53")}, false)
 		if err != nil {
-			return false, err
+			errs = append(errs, err)
+			continue
 		}
 
 		// NXDomain response is not really an error, just waiting for propagation to happen
 		if !(r.Rcode == dns.RcodeSuccess || r.Rcode == dns.RcodeNameError) {
-			return false, fmt.Errorf("NS %s returned %s for %s", ns, dns.RcodeToString[r.Rcode], fqdn)
+			errs = append(errs, fmt.Errorf("NS %s returned %s for %s", ns, dns.RcodeToString[r.Rcode], fqdn))
+			continue
```

Comment:
Looks like we're changing the behaviour here to try *each* NS instead of failing when one fails?

This function is called by checkDNSPropegation above (https://github.com/jetstack/cert-manager/blob/abfbb36a48e4ddea4e00c5df87cb26061a1013cc/pkg/issuer/acme/dns/util/wait.go#L77) which has the comment: `// checkDNSPropagation checks if the expected TXT record has been propagated to all authoritative nameservers.`

It seems like your change here would violate the above functions description.

Reply:
Ahhh, good catch. You are correct that I was changing the behavior because a) one of my servers as you may remember was a dead/AWOL server and the process would never complete. 

In my failure case, there wasn't enough information to determine why it was failing, so maybe if we kept the original behavior but logged more origin information, I would have caught that the dead server was in the SOA/NS RRs, fixed it and solved the issue.

Is this related to your mention on chat that all servers agreeing led to a higher probability of a remote server also agreeing? I'm poorly paraphrasing that chat here...

Reply:
Cool. Intuitively, seems like we could check all the servers and summarize the responses. For the untrained eye, the summary would capture the global view rather than having to parse where the retry begins and ends.

It will take a few days to get around to that, just wanted to keep a note here.